source/common/secret/secret_manager_impl.cc

```diff
@@ -6,23 +6,56 @@
 
 namespace Envoy {
 namespace Secret {
+namespace {
 
-void SecretManagerImpl::addOrUpdateSecret(const envoy::api::v2::auth::Secret& secret) {
+std::string configSourceHash(const envoy::api::v2::core::ConfigSource& config_source) {
```

Comment:
Should this be moved into https://github.com/envoyproxy/envoy/blob/master/source/common/config/utility.h or some other common location? Also, can we just use https://github.com/envoyproxy/envoy/blob/master/source/common/protobuf/utility.h#L141 here?

Reply:
Thanks for the advice! I will use MessageUtil::hash() to replace this hash function.